src/NServiceBus.Core/MessageHandlerContext.cs

```diff
@@ -0,0 +1,69 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NServiceBus.Transports;
+    using NServiceBus.Unicast;
+
+    class MessageHandlerContext : IMessageHandlerContext
+    {
+        ContextualBus bus;
+
+        public MessageHandlerContext(ContextualBus bus)
+        {
+            this.bus = bus;
+        }
+
+        public string MessageId => bus.MessageBeingProcessed.MessageId;
```

Comment:
@danielmarbach @andreasohlund I completely removed MessageContext and using the underlying `IncomingMessage` on the `ContextualBus` directly. What do you think?